File: examples/typescript/03-paginate-conversations.ts
// Walk every WhatsApp conversation on an agent. Pages are fetched on demand.
import { BimpeAI } from '@bimpeai/sdk';

const client = new BimpeAI({ apiKey: process.env.BIMPEAI_API_KEY ?? '' });
const agentId = process.argv[2];
if (!agentId) throw new Error('usage: 03-paginate-conversations.ts <agentId>');

let count = 0;
for await (const conversation of client.conversations.list(agentId, {
  channel: 'whatsapp',
  limit: 100,
})) {
  count += 1;
  if (count % 100 === 0) console.log(`seen ${count}`, conversation.id);
}
console.log(`total whatsapp conversations: ${count}`);

File: packages/sdk-python/examples/03_paginate_conversations.py
import os
import sys

from bimpeai import BimpeAI

if len(sys.argv) < 2:
    raise SystemExit("usage: 03_paginate_conversations.py <agent_id>")

client = BimpeAI(api_key=os.environ.get("BIMPEAI_API_KEY", ""))
count = 0
for conversation in client.conversations.list(sys.argv[1], channel="whatsapp", limit=100):
    count += 1
    if count % 100 == 0:
        print("seen", count, conversation.id)
print("total whatsapp conversations:", count)
